templates/crds/consoleservices.crd.yaml

```diff
@@ -60,6 +60,17 @@ spec:
               type: string
             host:
               type: string
+            impersonation:
+              type: object
+              description: "Enable impersonation as the mechanism for authorizing user actions. For impersonation to work, it is important to also grant the console-server permissions to impersonate users."
```

Comment:
How exactly is the ability to impersonate enabled?  My own curiosity and should we include a pointer to the relevant part of the kubernetes documentation in the EnMasse kubernetes docs?

Reply:
What I have not yet done is to add some docs on how to use this. To be able to impersonate, the console-server service-account must be granted 'impersonate' verb on a resource named 'users'. However, this permission is considered very broad, and not something I think we should add to the default permissions. 

I was thinking that having that role + rolebinding in a separate 'example-console' folder or something, and then refer to that in the documentation. Wdyt? Agreed on referring the kubernetes docs.

Reply:
@k-wall I've confirmed that this issue happens on OpenShift as well. If I create a user 'test1', and then apply the following as a cluster-admin user: 

```
---
apiVersion: v1
kind: Namespace
metadata:
  name: test1
---
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  labels:
    app: enmasse
  name: test1
rules:
  - apiGroups: [ "" ]
    resources: [ "namespaces" ]
    resourceNames: [ "test1" ]
    verbs: [ "get" ]
---
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
  name: test1
  labels:
    app: enmasse
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: test1
subjects:
- kind: User
  name: user1
  apiGroup: rbac.authorization.k8s.io
```

The 'test1' user is unable to see the namespace. However, if I modify the console-server to pass the Name in addition to Namespace, it works. I've raised a PR with a fix https://github.com/EnMasseProject/enmasse/pull/4133